src/applications/disability-benefits/all-claims/components/OfficialReport.jsx

```diff
@@ -0,0 +1,304 @@
+import React, { useEffect, useState, useRef } from 'react';
+import PropTypes from 'prop-types';
+import { SchemaForm } from 'platform/forms-system/exportsFile';
+import FormNavButtons from '~/platform/forms-system/src/js/components/FormNavButtons';
+import { useEditOrAddForm } from 'platform/forms-system/src/js/patterns/array-builder';
+import ArrayBuilderCancelButton from 'platform/forms-system/src/js/patterns/array-builder/ArrayBuilderCancelButton';
+import { getArrayUrlSearchParams } from 'platform/forms-system/src/js/patterns/array-builder/helpers';
+import {
+  VaModal,
+  VaButton,
+  VaAlert,
+} from '@department-of-veterans-affairs/component-library/dist/react-bindings';
+
+import {
+  officialReport,
+  officialReportMst,
+} from '../pages/form0781/officialReport';
+import { removePoliceReportModalContent } from '../content/officialReport';
+import { isRelatedToMST } from '../utils/form0781';
+import { POLICE_REPORT_LOCATION_FIELDS } from '../constants';
+
+export function shouldShowPoliceDataModal(event) {
+  return (
+    event &&
+    !event?.otherReports?.police &&
+    POLICE_REPORT_LOCATION_FIELDS.some(
+      field => typeof event[field] === 'string' && event[field].trim() !== '',
+    )
+  );
+}
+
+const OfficialReport = props => {
+  const formDomRef = useRef(null);
+  const modalRef = useRef(null);
+  const alertRef = useRef(null);
+  const submitButtonRef = useRef(null);
+
+  const searchParams = getArrayUrlSearchParams();
+  const isEdit = !!searchParams.get('edit');
+  const isAdd = !!searchParams.get('add');
+  const index = parseInt(props.pagePerItemIndex, 10);
+
+  const [tempData, setTempData] = useState(props.data || {});
+  const [showModal, setShowModal] = useState(false);
+  const [showAlert, setShowAlert] = useState(false);
+
+  const handlers = {
+    shouldShowPoliceDataModal,
+    onModalOrContinue: ({ formData }) => {
+      if (shouldShowPoliceDataModal(formData)) {
+        setShowModal(true);
+        return;
+      }
+
+      if (!formDomRef.current) return;
```

Comment:
Confused about what this is doing, maybe leave some docstrings in here for some of this more complex logic

Reply:
That's fair. I'll try to succinctly describe functionality and reasons for inclusion.